### DISCUSSION TITLE
Random syntax error

### DISCUSSION THREAD

**Discussion by Lia-gt** [unoplatform/uno#13779] [Q&A] [answered]
I suddenly started to get 2 syntax errors saying `Syntax error, ',' expected` in MainPage_e74fbf49b41a177d8d85219d73fa1ddd.cs. I can't click through to that file nor can I find it anywhere. Does anyone have any clue what to do about it?

**Comment by limeniye** [comment]
Hello, @delu-u.
To get more information and see the whole picture of your problem, please, send the MainPage.xaml file

**Reply by Lia-gt** [reply]
Here's the XAML. It started happening after I added access keys to the language menu, but stayed even after I removed them. 
```xaml
<Page x:Class="CetisDbMang.MainPage"
	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
	xmlns:local="using:CetisDbMang"
	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
	mc:Ignorable="d"
	Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
    <StackPanel HorizontalAlignment="Stretch" VerticalAlignment="Stretch">
        <MenuBar VerticalAlignment="Top">
            <MenuBarItem x:Uid="Entry" AccessKey="E">
                <MenuFlyoutItem Text="New" Click="NewEntry" AccessKey="N">
                    <MenuFlyoutItem.KeyboardAccelerators>
                        <KeyboardAccelerator Modifiers="Control" Key="N"/>
                    </MenuFlyoutItem.KeyboardAccelerators>
                </MenuFlyoutItem>
                <MenuFlyoutItem Text="Save" Click="SaveEntry" AccessKey="S">
                    <MenuFlyoutItem.KeyboardAccelerators>
                        <KeyboardAccelerator Modifiers="Control" Key="S"/>
                    </MenuFlyoutItem.KeyboardAccelerators>
                </MenuFlyoutItem>
                <MenuFlyoutItem Text="Delete" Click="DeleteEntry" AccessKey="D">
                    <MenuFlyoutItem.KeyboardAccelerators>
                        <KeyboardAccelerator Modifiers="Control" Key="Delete"/>
                    </MenuFlyoutItem.KeyboardAccelerators>
                </MenuFlyoutItem>
            </MenuBarItem>
            <MenuBarItem x:Uid="Lang">
                <MenuFlyoutItem Text="SK" Click="ChangeLang" Tag="sk"/>
                <MenuFlyoutItem Text="EN" Click="ChangeLang" Tag="en"/>
            </MenuBarItem>
        </MenuBar>
        <Grid>
            <Grid.ColumnDefinitions>
                <ColumnDefinition Width="*"/>
                <ColumnDefinition Width="*"/>
            </Grid.ColumnDefinitions>
            <StackPanel Grid.Column="0">
                <Expander x:Uid="Destinations" HorizontalAlignment="Stretch">
                    <ListView x:Name="DestinationsList" ItemClick="LoadDestinationEntry"/>
                </Expander>
                <Expander x:Uid="Stops" HorizontalAlignment="Stretch">
                    <ListView x:Name="StopsList" ItemClick="LoadStopEntry"/>
                </Expander>
                <Expander x:Uid="Trips" HorizontalAlignment="Stretch">
                    <ListView x:Name="TripsList" ItemClick="LoadTripEntry"/>
                </Expander>
                <Expander x:Uid="Shifts" HorizontalAlignment="Stretch">
                    <ListView x:Name="ShiftsList" ItemClick="LoadShiftEntry"/>
                </Expander>
                <Expander Padding="5" x:Uid="Drivers" HorizontalAlignment="Stretch" HorizontalContentAlignment="Stretch">
                    <ListView x:Name="DriversList" ItemClick="LoadDriverEntry">
                    </ListView>
                </Expander>
                <Expander x:Uid="Fonts" HorizontalAlignment="Stretch">
                    <ListView x:Name="FontsList" ItemClick="LoadFontEntry"/>
                </Expander>
            </StackPanel>
        </Grid>
    </StackPanel>
</Page>
```

**Reply by limeniye** [reply]
It works good for me. Please, attach a sample with this problem.
Also, try to clean + rebuild.

**Reply by Lia-gt** [reply]
I've tried to clean and rebuild multiple times already with no luck. This is what happens any time I try to run the app. I'm pretty sure the first error is a ghost error since it only started to appear after cleaning and rebuilding again and because it appeared multiple times before with the app running without any problems.
![image](https://github.com/unoplatform/uno/assets/95361071/6412382b-7b95-4b20-b4f5-ff28955284d4)

**Reply by limeniye** [reply]
Ok, lets see what is the problem are. Please, attach the .zip file with your solution.

**Reply by Lia-gt** [reply]
Oh one thing I forgot to mention. When rebuilding I get errors saying there are missing dlls in the MacOs project. Here's a link for the zip on my drive cuz it's too big to fit here: [https://drive.google.com/file/d/18h7ToEwh3qGyRlYcftvSsADgYtNL-43Q/view?usp=sharing](url)

**Reply by limeniye** [reply]
To fix this problem open your **en\Resources.resw** file as Notepad and comment the following resources:
**Entry.Title**,  **Lang.Title**
```XML
  <!-- <data name="Entry.Title" xml:space="preserve"> -->
    <!-- <value>Entry</value> -->
  <!-- </data> -->
  <!-- <data name="Lang.Title" xml:space="preserve"> -->
    <!-- <value>Language</value> -->
  <!-- </data> -->
```
Open your **Directory.Build.props** file and change ```DefaultLanguage``` to **en**

Currently I don't know why but its crushing your application.
Maybe @jeromelaban has already seen something similar.

**Reply by limeniye** [reply, marked as answer]
It looks like it's already reserved names. For example: with **Lang.Title** it doesn't work but with **Lang.Titlee** it works well.

**Reply by Lia-gt** [reply]
So basically it's a naming issue. 🙃 Why does it need that extra e at the end tho. That's weird.

**Reply by limeniye** [reply]
Basically, yes. Your main problem is translation resources.
Also is interesting that if you delete **Entry.Title, Lang.Title** for **sk** it still won't work. But with **en** its will.

Also I saw that your sk-file encoding is different than en-file:

![image](https://github.com/unoplatform/uno/assets/69314237/0b5fc350-49f0-44a5-8b97-6cf7943f5129)

<h3>Second way to fix this problem</h3>

Also you can move your menu items to **MenuBar.Items**:
```XML
<MenuBar>
  <MenuBar.Items>
      <-- <MenuFlyoutItem ... -->      
  </MenuBar.Items>
</MenuBar>
```

I saw that @dansiegel alredy worked with something similar. Maybe he knows the solution.

**Reply by Lia-gt** [reply]
Alrighty. Thanks for the help.
